### PR TITLE
Expose number of read-only shards because of write-concern

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

This PR adds a metric, that counts the number of shards per database that are currently in read-only mode, because there are not enough in-sync followers (less than write concern).

### CODE REVIEW

File: arangod/Metrics/InstrumentedBool.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  InstrumentedBool& operator=(bool value) {
+    set(value);
+    return *this;
+  }
+
+  void set(bool value) {
+    if (value == _value) {
+      return;
+    }
+    _value = value;
```
Comment (jsteemann) [inline arangod/Metrics/InstrumentedBool.h:61]: Does this part need to be thread-safe?
What if different threads try to push concurrently to the same InstrumentedBool instance?

Comment (maierlars) [reply arangod/Metrics/InstrumentedBool.h:61]: It is not thread safe, similar to any other `bool` variable.

Comment (jsteemann) [reply arangod/Metrics/InstrumentedBool.h:61]: Ok, thanks for confirming. It think it is good as it is.

**Review by jsteemann — APPROVED** [review]
LGTM

**Pull request merged**